src/dune_pkg/resolve_opam_formula.mli

```diff
@@ -0,0 +1,63 @@
+open! Import
+
+(** Update a filter by substituting variables from a [Solver_env.t] *)
+val substitute_variables_in_filter
+  :  ?stats_updater:Solver_stats.Updater.t
+  -> Solver_env.t
+  -> OpamTypes.filter
+  -> OpamTypes.filter
+
+(** Evaluate the filters in a filtered formula returning the resulting formula. *)
+val filtered_formula_to_formula
+  :  ?stats_updater:Solver_stats.Updater.t
```

Comment:
Unless, there's a specific need we try to avoid optional arguments. They introduce bugs, and their main utility which is improving an API in a backwards compatible way isn't relevant to our internal code.